smif/convert/interval.py

```diff
@@ -462,71 +437,67 @@ def _check_interval_in_register(self, interval):
             msg = "The interval set '{}' is not in the register"
             raise ValueError(msg.format(interval))
 
-    def convert(self, timeseries, from_interval, to_interval):
+    def convert(self, data, from_interval_set_name, to_interval_set_name):
         """Convert some data to a time_interval type
 
         Parameters
         ----------
-        timeseries: :class:`~smif.convert.interval.TimeSeries`
+        data: :class:`numpy.ndarray`
             The timeseries to convert from `from_interval` to `to_interval`
-        from_interval: str
+        from_interval_set_name: str
             The unique identifier of a interval type which matches the
             timeseries
-        to_interval: str
+        to_interval_set_name: str
             The unique identifier of a registered interval type
 
         Returns
         -------
-        dict
-            A dictionary with keys `name` and `value`, where the entries
-            for `key` are the name of the target time interval, and the
-            values are the resampled timeseries values.
+        :class:`numpy.ndarray`
+            An array of the resampled timeseries values.
 
         """
-        results = []
+        from_interval_set = self.get_intervals_in_set(from_interval_set_name)
+        to_interval_set = self.get_intervals_in_set(to_interval_set_name)
 
-        self._check_interval_in_register(from_interval)
-        self._convert_to_hourly_buckets(timeseries, from_interval)
+        converted = np.zeros(len(to_interval_set))
+        hourly_values = self._convert_to_hourly_buckets(data, from_interval_set)
 
-        target_intervals = self.get_intervals_in_set(to_interval)
-        for name, interval in target_intervals.items():
-            self.logger.debug("Resampling to %s", name)
+        for idx, interval in enumerate(to_interval_set.values()):
             interval_tuples = interval.to_hours()
 
-            total = 0
-
             for lower, upper in interval_tuples:
                 self.logger.debug("Range: %s-%s", lower, upper)
-                total += sum(timeseries.hourly_values[lower:upper])
-
-            results.append({'id': name,
-                            'value': total})
+                converted[idx] += sum(hourly_values[lower:upper])
 
-        return results
+        return converted
 
     def _convert_to_hourly_buckets(self, timeseries, interval_set):
         """Iterates through the `timeseries` and assigns values to hourly buckets
 
         Parameters
         ----------
-        timeseries: :class:`~smif.convert.interval.TimeSeries`
+        timeseries: :class:`numpy.ndarray`
```

Comment:
Instead of converting all values to hourly buckets, this could move to the coefficient generation used in `smif.area.RegionRegister`, building coefficient matrices (which are held in memory) across interval definitions.  This will then enable quick vectorised conversions at runtime...  We should monitor memory use here.